### DISCUSSION TITLE
Default sort does not change the matching column

### DISCUSSION THREAD

**Discussion by sinnbeck** [filamentphp/filament#7781] [Help] [answered]
I am setting `->defaultSort('created_at', 'desc')` on a table. But I also have that same column as sortable
```php
TextColumn::make('created_at')
                    ->sortable()
                    ->dateTime()
```
But filament does not set the actual column as sorted, so if I click it, it just sets the same order as it already has, and changes the arrow. Is there a way to actually force the column header to update as sorted?

**Comment by sinnbeck** [comment, marked as answer]
I was able to trick it by adding these. I would think they might be nice to add to filament itself :)
```php
 public function mount()
    {
        $this->tableSortColumn = 'created_at';
        $this->tableSortDirection = 'desc';
    }
```
Only problem now is that filament uses 3 state sort, which means clicking the column turns off sorting, which gives the exact result. Both ways make the sorting seem broken

**Reply by sinnbeck** [reply]
I overwrote the table sort method as well
```php
 public function sortTable(?string $column = null, ?string $direction = null): void
    {
        if ($column === $this->tableSortColumn) {
            $direction ??= match ($this->tableSortDirection) {
                'asc' => 'desc',
                'desc' => 'asc',
                default => 'asc',
            };
        } else {
            $direction ??= 'asc';
        }

        $this->tableSortColumn = $column;
        $this->tableSortDirection = $direction;

        $this->updatedTableSortColumn();
    }

**Comment by danharrin** [comment]
The arrow is deliberately not activated when you have a default sort applied for that column. I can't recall all the situations off the top of my head, but we have done extensive testing and had a lot of user feedback and this is the way that worked the best across many use cases.

**Reply by sinnbeck** [reply]
That's fair. I was able to work around it myself, to avoid confusion :)
